app/models/user.rb

```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  has_many :posts, dependent: :destroy
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def recent_posts(limit = 3)
+    posts.order(created_at: :desc).limit(limit)
+  end
+
+  def update_posts_counter
+    update(posts_counter: posts.count)
+  end
```

Comment:
- [ ] Kindly remove this method. It is meant to the written inside the post model.